src/rust/storage/seg/src/segments/header.rs

```diff
@@ -0,0 +1,276 @@
+    }
+
+    #[inline]
+    /// Decrement the number of live items.
+    pub fn decr_live_items(&mut self) {
+        self.live_items -= 1;
+    }
+
+    #[inline]
+    /// Returns the TTL for the segment.
+    pub fn ttl(&self) -> CoarseDuration {
+        CoarseDuration::from_secs(self.ttl)
+    }
+
+    #[inline]
+    /// Sets the TTL for the segment.
+    pub fn set_ttl(&mut self, ttl: CoarseDuration) {
+        self.ttl = ttl.as_secs();
+    }
+
+    #[inline]
+    /// The number of bytes used in the segment.
+    pub fn live_bytes(&self) -> i32 {
+        self.live_bytes
+    }
+
+    #[inline]
+    /// Increment the number of bytes used in the segment.
+    pub fn incr_live_bytes(&mut self, bytes: i32) -> i32 {
+        let prev = self.live_bytes;
+        self.live_bytes += bytes;
+        prev
+    }
+
+    #[inline]
+    /// Decrement the number of bytes used in the segment.
+    pub fn decr_live_bytes(&mut self, bytes: i32) -> i32 {
+        let prev = self.live_bytes;
+        self.live_bytes -= bytes;
+        prev
+    }
+
+    #[inline]
+    /// Returns an option containing the previous segment id if there is one.
+    pub fn prev_seg(&self) -> Option<NonZeroU32> {
+        self.prev_seg
+    }
+
+    #[inline]
+    /// Set the previous segment to some id. Passing a negative id results in
+    /// clearing the previous segment pointer.
+    pub fn set_prev_seg(&mut self, id: Option<NonZeroU32>) {
+        self.prev_seg = id;
+    }
+
+    #[inline]
+    /// Returns an option containing the next segment id if there is one.
+    pub fn next_seg(&self) -> Option<NonZeroU32> {
+        self.next_seg
+    }
+
+    #[inline]
+    /// Set the next segment to some id. Passing a negative id results in
+    /// clearing the next segment pointer.
+    pub fn set_next_seg(&mut self, id: Option<NonZeroU32>) {
+        self.next_seg = id;
+    }
+
+    #[inline]
+    /// Returns the instant at which the segment was created
+    pub fn create_at(&self) -> CoarseInstant {
+        self.create_at
+    }
+
+    #[inline]
+    /// Update the created time
+    pub fn mark_created(&mut self) {
+        self.create_at = CoarseInstant::recent();
+    }
+
+    #[inline]
+    /// Returns the instant at which the segment was merged
+    pub fn merge_at(&self) -> CoarseInstant {
+        self.merge_at
+    }
+
+    #[inline]
+    /// Update the created time
+    pub fn mark_merged(&mut self) {
+        self.merge_at = CoarseInstant::recent();
+    }
+
+    #[inline]
+    // clippy throws a false positive for suspicious_operation_groupings lint
+    // for the instant + duration portion. We set the allow pragma to silence
+    // the false positive.
+    #[allow(clippy::suspicious_operation_groupings)]
+    /// Can the segment be evicted?
+    pub fn can_evict(&self) -> bool {
+        self.evictable()
+            && self.next_seg().is_some()
+            && (self.create_at() + self.ttl()) >= (Instant::recent() + Duration::from_secs(5))
```

Comment:
Does the number 5 have any specific meaning?

Reply:
This was taken from a previous version of the C implementation - looks like this got moved to an option `seg_mature_time` with a default value of 20s. I'll parameterize this.

Reply:
This is actually a little tricky to parameterize, I've moved the definition into a constant and added a TODO to parameterize it.

LMK if this is sufficient or if I should address this further in this PR.

Reply:
sounds good